addons/project_timesheet_holidays/models/res_config_settings.py

```diff
@@ -1,11 +1,16 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import fields, models
+from odoo import api, fields, models
 
 
 class ResConfigSettings(models.TransientModel):
     _inherit = 'res.config.settings'
 
-    leave_timesheet_project_id = fields.Many2one(related='company_id.leave_timesheet_project_id', string="Internal Project", domain="[('company_id', '=', company_id)]", readonly=False)
+    leave_timesheet_project_id = fields.Many2one(related='company_id.leave_timesheet_project_id', required=True, string="Internal Project", domain="[('company_id', '=', company_id)]", readonly=False)
     leave_timesheet_task_id = fields.Many2one(related='company_id.leave_timesheet_task_id', string="Time Off Task", domain="[('company_id', '=', company_id)]", readonly=False)
+
+    @api.onchange('leave_timesheet_task_id')
+    def _onchange_leave_timesheet_task_id(self):
```

Comment:
We avoid to add onchanges if it's possible to achieve the same thing with a computed stored field instead. So we should here make the field a computed/stored/editable field with a compute method and an inverse method.

On the other hand, here you modify the related field, so you don't edit the current record, but the related record. This is often a bad practice because this will impact a lot of records. But here I supposed it should be ok.